src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/SessionSummaryService.kt

```diff
@@ -99,49 +103,79 @@ class SessionSummaryService(
   private fun getSessionSummaries(
     prisonerNumbers: List<String>,
   ): SessionSummaries {
+    val today = LocalDate.now()
+
     val sessionSummaries = SessionSummaries()
     val inductionSchedules = inductionScheduleService.getInCompleteInductionSchedules(prisonerNumbers)
     val reviewSchedules = reviewScheduleService.getInCompleteReviewSchedules(prisonerNumbers)
 
     inductionSchedules.forEach { schedule ->
       when {
-        schedule.scheduleStatus.includeExceptionOnSummary() -> {
-          sessionSummaries.exemptInductions.add(schedule)
-        }
-        schedule.scheduleStatus == INDUCTION_SCHEDULED && schedule.deadlineDate < today -> {
-          sessionSummaries.overdueInductions.add(schedule)
-        }
-        schedule.scheduleStatus == INDUCTION_SCHEDULED &&
-          today in schedule.deadlineDate.minusMonths(2)..schedule.deadlineDate -> {
-          sessionSummaries.dueInductions.add(schedule)
-        }
+        schedule.includeInExemptCount() -> sessionSummaries.exemptInductions.add(schedule)
```

Comment:
Refactored the logic into private extension functions to make it easier to grok this code